src/google/protobuf/parse_context.h

```diff
@@ -929,15 +939,60 @@ template <typename T>
     return p + 2;
   }
   return VarintParseSlowArm(p, out, first8);
-#else   // __aarch64__
+#elif defined(__x86_64__) // __aarch64__
```

Comment:
Is this optimization meant for x86_64 or arm64?  If it's x86_64, the build error is because you're using ValueBarrier, which we only define for arm64 builds.  If it's for arm64, you need to fix this line and probably call that out in the PR description.  We tend to be willing to accept a lot of ugly code for x86_64 optimization that deliver, but for arm64 the bar would likely be higher

Reply:
Aha. This is an optimization for x86_64. I'll fix that soon.

Reply:
Cool, also note that this varint parser is not particularly load-bearing anymore.  ShiftMixParseVarint is our table-driven parser's implementation that's used for parsing most protobuf data on the wire.

Reply:
I'm aware. I tried to look into if this would he helpful for table parsing but concluded it wasn't. I was still surprised it managed an extra 4%.

A little unrelated but I was writing a protobuf library in Rust which uses this in a load-bearing manner :)